cmd/juju/commands/helptool_test.go

```diff
@@ -32,15 +34,36 @@ func (suite *HelpToolSuite) TestHelpTool(c *gc.C) {
 	expectedNames := jujuc.CommandNames()
 	output := badrun(c, 0, "help-tool")
 	lines := strings.Split(strings.TrimSpace(output), "\n")
+	template := "%v"
+	if runtime.GOOS == "windows" {
+		template = "%v.exe"
+	}
 	for i, line := range lines {
-		lines[i] = strings.Fields(line)[0]
+		command := strings.Fields(line)[0]
+		lines[i] = fmt.Sprintf(template, command)
 	}
-	if runtime.GOOS == "windows" {
-		for i, command := range lines {
-			lines[i] = command + ".exe"
+	c.Assert(lines, gc.DeepEquals, expectedNames)
+}
+
+// Component-based features such as payloads and resources
+// are different enough in implementation to the rest
+// of Juju code that we need to ensure that help-tool can reach them
+// explicitely.
```

Comment:
I was wondering that. I'll change and will hard code the list.It'll also reduce number of tests back down :D